### DISCUSSION TITLE
Trying to pass "markup" value in REST body as raw/json

### DISCUSSION THREAD

**Discussion by sid6p8** [Budibase/budibase#5297] [Help]
Hello,

I'm trying to create a REST query where I post some values in the body:

![image](https://user-images.githubusercontent.com/51316894/162376102-76fb68a8-f99f-4926-afb0-a19063a5a228.png)

When I try to run the query I get a budibase error: Error running query

It seems to be related to the last value in the body, where the key is "markup". If I remove this, or change it to anything else, e.g. "markups", the query gets executed. The server doesn't like it, but that's to be expected if the keys are wrong.

Any idea what might cause this? Or do you know a workaround? Did I miss something?
Thanks!

Regards
Ferdinand

**Comment by melohagan** [comment]
Hey @sid6p8 

Would you mind showing me the error in the network tab please? F+12 on Chrome.

**Reply by sid6p8** [reply]
Sure thing:
![image](https://user-images.githubusercontent.com/51316894/162252898-67667975-f562-4005-b5d0-fb5374425154.png)

[post.txt](https://github.com/Budibase/budibase/files/8444936/post.txt)

I think I got all the data (Save all as HAR), but let me know if you need more or something else.

Cheers!

**Reply by melohagan** [reply]
Thanks @sid6p8 

What are you expecting to be returned from the wikie2html endpoint? 
Maybe try adding the header: Accept: text/html

**Reply by sid6p8** [reply]
Hey @melohagan,

I tried to get the query running with postman in parallel, and got it to work with these keys/values. The response I get with postman is:
`<span class='wikiContect'>Hello <b>World</b>!</span>`

Adding an accept: text/html header in budibase doesn't help, though.

What got me confused is that with this exact query I get the "Error running query" error, while with almost everything else I get a query sent successfully, but see the result stating, that there was an issue. So I thought that budibase didn't even send out a query, but it did.

I tried it in the most primitive way possible with curl:
`fg@IT03xxx:~$ curl -X POST "https://cxxx.yyyy.com:1234/cb/api/v3/projects/76/wiki2html" \`
`>   -H "accept: application/json" \`
`>   -H "Authorization: Basic Zmdyb3NzbWFubjpGR04yMDIxIQ==" \`
`>   -H "Content-Type: application/json" \`
`>   --data-raw "{\"contextId\": 14384, \"contextVersion\": 10, \"renderingContextType\": \"TRACKER_ITEM\", \"markup\": \"Hello __World__\"}"`

`<span class='wikiContent'>Hello <b>World</b></span>`

I've been unable to figure out what's different to the budibase request. Is there an easy way to see the request body? Looking at the info from Chrome I think it is

`$'{"_id":"query_datasource_54aa16e75d984111847fda590bb31208_968a0d5d2d244686a72a398798a6ba96","_rev":"19-9a1927442babee5b6e3dde5126d52bdc","datasourceId":"datasource_54aa16e75d984111847fda590bb31208","name":"wiki2html","parameters":{"wikitext":"Text in wiki format to convert to HTML","itemId":"14384","itemVersion":"10","projectId":"76"},"fields":{"headers":{"Accept":"text/html"},"queryString":"","path":"https://codebeamer.duagon.com:8443/cb/api/v3/projects/76/wiki2html","pagination":{"type":null},"bodyType":"json","enabledHeaders":{},"disabledHeaders":{"Accept":false},"authConfigId":"ce2f5af12ee0b4af7b1c8303a89763f1e","requestBody":"{\\n    \\"contextId\\": 14384,\\n    \\"contextVersion\\": 10,\\n    \\"renderingContextType\\": \\"TRACKER_ITEM\\",\\n    \\"markup\\": \\"Hello __World__\u0021\\"\\n}"},"transformer":"return data","schema":{"message":{"name":"message","type":"string"},"resourceUri":{"name":"resourceUri","type":"string"}},"readable":true,"queryVerb":"create","flags":{"urlName":false}}'`

...but that seems a bit much.

Any idea what to check or where to look?

Thanks you!

**Reply by melohagan** [reply]
Hmm, I wonder if the **\u0021** is causing problems. If you remove the exclamation mark from the "markup" value does the query execute?

**Reply by sid6p8** [reply]
Good idea, but sadly: no. Doesn't change anything.

**Reply by sid6p8** [reply]
I managed to capture the ethernet traffic between the budibase server and the endpoint I'm trying to reach and it looked as if the request body was not "escaped"(?), i.e. `"` instead of `\"`.

So I tried setting the body to raw(Text) and added the required escape characters. That didn't work because the content-type was automatically set to text/plain, which the server doesn't seem to like.

Setting the body back to raw(JSON) and escaping the text manually leads to the query not being executed on the server, that is, no query to the actual endpoint is generated.

Any ideas?
